warp_pipe.go

```diff
@@ -165,20 +165,26 @@ func (w *WarpPipe) Close() error {
 }
 
 // IsLatestChangeSet returns true if the id argument matches that of the last record in the changeset table.
+// TODO: This feature only supports the notify listener. It needs to support others.
 func (w *WarpPipe) IsLatestChangeSet(id int64) (bool, error) {
-	rows, err := w.conn.Query("SELECT id FROM warp_pipe.changesets ORDER BY id DESC LIMIT 1")
-	if err != nil {
-		return false, fmt.Errorf("failed to query latest changeset record: %w", err)
-	}
-	for rows.Next() {
-		latestID := int64(0)
-		err := rows.Scan(&latestID)
+	switch w.listener.(type) {
```

Comment:
I just have this one question: Why is the check for listener being done for every changeset? Was there any issues with panic/error early if the listener is on an unsupported tyoe?

Reply:
I think this is fine for now. The logic needs to change as part of the TODO. It should fail in the function, since thats where the dependency is imo.